chunkserver/src/main/java/com/example/chunkserver/ChunkServerController.java

```diff
@@ -1,29 +1,73 @@
 package com.example.chunkserver;
+
+import com.example.chunkserver.entity.Chunk;
+import jakarta.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 @RestController
 @RequestMapping("/chunkserver")
 public class ChunkServerController {
 
-    private final Map<String, byte[]> chunkStorage = new HashMap<>(); // Chunk ID to Data
+    @Autowired
+    private ChunkServerService chunkServerService;
+    private Map<String, List<Chunk>> chunkStorage = new HashMap<>();
+
+    @PostConstruct
+    public void initializeChunkStorage() {
+        System.out.println("Initializing chunk storage...");
+        chunkStorage = chunkServerService.retrieveChunks();
+        System.out.println("Chunk storage initialized: " + chunkStorage);
+    }
 
     @PostMapping("/storeChunk")
-    public ResponseEntity<String> storeChunk(@RequestParam String chunkId, @RequestBody byte[] data) {
-        chunkStorage.put(chunkId, data);
-        return ResponseEntity.ok("Chunk stored: " + chunkId);
+    public ResponseEntity<String> storeChunk(@RequestParam String filename, @RequestBody Chunk chunk) {
+        System.out.println("Received request to store chunk: " + chunk.getId() + ", for file: " + filename);
+        List<Chunk> chunks = chunkStorage.containsKey(filename) ? chunkStorage.get(filename) : new ArrayList<>();
+
+        chunks.stream()
+                .filter(c -> c.getId().equals(chunk.getId()))
+                .findFirst()
+                .ifPresentOrElse(
+                        existingChunk -> chunks.set(chunks.indexOf(existingChunk), chunk),
+                        () -> chunks.add(chunk)
+                );              //add new chunk, or replace if same chunkId exists already
+
+        chunkStorage.put(filename, chunks);
+        chunkServerService.storeChunk(filename, chunk);
+        System.out.println("Stored ChunkID: " + chunk.getId() + ", for File: " + filename);
+        System.out.println("Current Chunks stored for file: "+filename);
+        for (Chunk c : chunks){
+            System.out.println(c.getId());
+        }
+        return ResponseEntity.ok("Stored ChunkID: " + chunk.getId() + ", for File: " + filename);
```

Comment:
Can you handle error scenarios?. I don't see errors being returned here in case the file is not stored